src/ray/raylet/node_manager.cc

```diff
@@ -618,10 +699,19 @@ void NodeManager::ProcessClientMessage(
       RAY_CHECK(!driver_id.is_nil());
       local_queues_.RemoveDriverTaskId(driver_id);
       worker_pool_.DisconnectDriver(driver);
+
+      RAY_LOG(DEBUG) << "Driver (pid=" << driver->Pid() << ") is disconnected. "
+                     << "driver_id: " << driver->GetAssignedDriverId();
     }
     return;
   } break;
   case protocol::MessageType::SubmitTask: {
+    std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
+    if (worker && worker->IsDead()) {
```

Comment:
Can we move the worker dead detection in the worker_pool_.GetRegisteredWorker?

Reply:
There are a few places that use if '''(worker)''' to determine whether this is a worker, it it's nullptr then it's considered a writer.  With that I think it's probably better to keep the current logic.